src/hw/deviceAccess.ts

```diff
@@ -88,12 +88,14 @@ export const withDevice =
       let sub;
       const deviceQueue = deviceQueues[deviceId] || Promise.resolve();
 
-      const finalize = (transport, cleanups) =>
-        close(transport, deviceId)
+      const finalize = (transport, cleanups) => {
+        transport.setAllowAutoDisconnect(true);
```

Comment:
not all transport have this method 🤔 sounds like it will break for some transport implementation then. i'm also curious why it passes the type system too